### PR TITLE
8305505: NPE in javazic compiler

### PR DESCRIPTION
**Pull request by satoyoshiki**
Please review this PR.  
With this minor change, the javazic compiler (Main.java) can produce the HTML files that display given time zone data correctly.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8305505](https://bugs.openjdk.org/browse/JDK-8305505): NPE in javazic compiler

### Reviewers
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13504/head:pull/13504` \
`$ git checkout pull/13504`

Update a local copy of the PR: \
`$ git checkout pull/13504` \
`$ git pull https://git.openjdk.org/jdk.git pull/13504/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13504`

View PR using the GUI difftool: \
`$ git pr show -t 13504`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13504.diff">https://git.openjdk.org/jdk/pull/13504.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13504#issuecomment-1521169847)

### CODE REVIEW

File: test/jdk/sun/util/calendar/zi/GenDoc.java [hunk 1]
```diff
@@ -154,24 +154,26 @@ int processZoneinfo(Timezone tz) {
...
-                BufferedReader in = new BufferedReader(fr);
-                mapList = new HashMap<String,LatitudeAndLongitude>();
-                String line;
-                while ((line = in.readLine()) != null) {
-                    // skip blank and comment lines
-                    if (line.length() == 0 || line.charAt(0) == '#') {
-                        continue;
+            if (mapList == null) {
+                mapList = new HashMap<String, LatitudeAndLongitude>();
+                if (Main.getMapFile() != null) {
```
Comment (naotoj) [inline test/jdk/sun/util/calendar/zi/GenDoc.java:159]: You could retain the map file and use it in the following line

Comment (satoyoshiki) [reply test/jdk/sun/util/calendar/zi/GenDoc.java:159]: Correct.  Thanks for the catch.

**Review by naotoj — COMMENTED** [review]
I revisited the code, and now think that `mapList` better be `null`, as it indicates the map option was not provided, so I think modifying the offending line as
```
            LatitudeAndLongitude location = (mapList ! = null ? mapList.get(zonename) : null);
```
may be cleaner.

**Review by naotoj — APPROVED** [review]

**Pull request closed without merging**